packages/otelbin/src/components/share/DownloadButton.tsx

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2023 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { useReactFlow, getRectOfNodes, getTransformForBounds } from 'reactflow';
+import { ImageDown } from "lucide-react";
+import { Button } from "~/components/button";
+import { toPng } from 'html-to-image';
+
+const downloadImage = (dataUrl: string) => {
+  const a = document.createElement('a');
+
+  a.setAttribute('download', 'collector-image.png');
+  a.setAttribute('href', dataUrl);
+  a.click();
+};
+
+const calculateImageDimensions = (nodeCount: number) => {
+  const baseWidth = 1024;
+  const baseHeight = 768;
+  const widthPerNode = 100; 
+
+  const calculatedWidth = Math.max(baseWidth, widthPerNode * nodeCount);
```

Comment:
**Issue:** This is an approximation that is not likely to be correct. E.g. for many pipelines this will result in the wrong image dimensions.

It would be better if we could find a way to work with the ReactFlow API to identify the dimensions / scale it appropriately. Some thoughts: Could we use [getNodesBounds](https://reactflow.dev/api-reference/utils/get-nodes-bounds) to identify the required space?

Reply:
And then for example transform based on the horizontal space required for one node (as that is fixed)?

Reply:
Make sense to me. I just use it because I tried to use this method and I didn't find, it was necessary to update the library to it haha. 
Yeah, it was an approximation because the previous methods to get the Bounds `getReactNodes` isn't work properly to get the right values, for some tests that I did the dimensions / scale were greater than necessary. But, the new method on the new version is work great. I'll commit it.